### ISSUE TITLE
XMLHttpRequest() with custom user-agent is ignored

### ISSUE THREAD

**Issue by cibersheep** [ubports/ubuntu-touch#1508]
### Description of the feature
Trying to set a custom user-agent using a request like this in qml will fail (returning OS  user-agent)
```
    var xhr = new XMLHttpRequest();
    xhr.open("POST", 'https://www.whatismyip.com/user-agent-info/', true);
    xhr.setRequestHeader("User-Agent", "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0")
    xhr.send();
```

- Device: all
- Channel: all

**Comment by lduboeuf** [comment]
Don't know if Qt has changed since:
https://bugreports.qt.io/plugins/servlet/mobile#issue/QTBUG-20473

**Comment by cibersheep** [comment]
I'm not completely sure but might be changed:
https://fetch.spec.whatwg.org/#default-user-agent-value

I can't find more info but if it would be forbidden, would they appear here, right?
https://fetch.spec.whatwg.org/#forbidden-header-name

Could be unrelated but you can set it in firefox command line

**Comment by UniversalSuperBox** [comment]
Is it possible to send a request with a different user-agent from JavaScript in Firefox? I don't think you're allowed to do that.

**Comment by cibersheep** [comment]
> Is it possible to send a request with a different user-agent from JavaScript in Firefox? I don't think you're allowed to do that.

Yes. It is possible, you need to get around CORS error but setting header "User-Agent" is valid.

**Comment by UniversalSuperBox** [comment]
Aha, yep, it's explicitly denied in Qt: https://github.com/qt/qtdeclarative/blame/v5.9.5/src/qml/qml/qqmlxmlhttprequest.cpp#L1810

You can see here that this code has not been changed for over 10 years. It's not allowed to modify some headers of the XmlHttpRequest, it simply ignores your call.

I've asked in #qt on freenode if this is on purpose, we'll see if I get an answer.

**Comment by cibersheep** [comment]
Interesting. Thanks!
